### PR TITLE
feat(bootstrap): template mirror + name substitution + git init (#204)

### LINKED ISSUE #204: [#156 sub-B] template mirror + name substitution + cross-repo loop update
Sub-issue of #156 (`scripts/bootstrap-new-repo.sh` setup wizard).

## Scope

Ship `scripts/bootstrap/template-mirror.sh`: the rsync-with-excludes step that copies Mergepath into the new repo, the deterministic file substitution for the 6 name-bearing files, and the cross-repo loop update that adds the new repo's name to Mergepath's own `DEPLOYMENT.md` + `REVIEW_POLICY.md`.

Depends on #156 sub-A (wizard scaffold provides the dispatch shape and `bootstrap::run` wrapper this sub-issue's stage plugs into).

### 1. Template mirror (step 3 of #156's flow)

```bash
bootstrap::stage_template_mirror() {
  local target="$HOME/GitHub/$NEW_REPO"
  bootstrap::run "rsync mergepath → $target" \
    rsync -a "$MERGEPATH_ROOT/" "$target/" \
    --exclude='.git/' \
    --exclude='.DS_Store' \
    --exclude='dist/' \
    --exclude='mergepath/' \
    --exclude='packaging/' \
    --exclude='.claude/worktrees/' \
    --exclude='.claude/settings.local.json' \
    --exclude='.claude/launch.json' \
    --exclude='specs/mergepath_playground.md' \
    --exclude='plans/mergepath-playground.md' \
    --exclude='scripts/policy-sim.sh' \
    --exclude='bugs/screenshots/' \
    --exclude='.github/screenshots/'

  # The exclude list leaves the playground test file behind; remove
  # it explicitly. Same for any other "left-behind orphans" identified
  # during matchline bootstrap.
  rm -f "$target/tests/test_mergepath_playground.sh"
  rmdir "$target/bugs" 2>/dev/null || true  # only if now-empty
}
```

The exclude list lives in a single source of truth. If a future Mergepath PR adds a new mergepath-only file, the bootstrap test (below) catches the omission.

### 2. Name substitution (step 4 of #156's flow)

For each of the 6 name-bearing files identified in #156's Phase B walkthrough:

```bash
NAME_BEARING_FILES=(
  README.md
  BRAND.md
  .ai_context.md
  docs/agents/repository-overview.md
  .repo-template.yml
  SECURITY.md
)
```

Apply substitutions:
- `{{REPO_NAME}}` → `$NEW_REPO` (lowercase, hyphenated)
- `{{REPO_DESCRIPTION}}` → `$NEW_REPO_DESCRIPTION` (the prompt input)
- `{{REPO_URL}}` → `https://github.com/nathanjohnpayne/$NEW_REPO`

The substitution markers don't currently exist in mergepath's templates — this sub-issue adds them. Two approaches:

a. **Marker-driven (preferred)**: edit each of the 6 files in mergepath itself to introduce `{{REPO_NAME}}`-style markers wherever "mergepath" appears in a name-bearing context (NOT in historical incident references; see #156's "Left intentionally unchanged" list). Then the wizard's substitution is a literal text replace.

b. **Heuristic**: leave mergepath's files as-is and have the wizard `sed` `'mergepath'` → `$NEW_REPO` with a curated allow-list of files. Brittle; the marker approach is cleaner.

Going with (a). The marker introduction PR is part of this sub-issue.

### 3. Drop mergepath-specific config (step 3 cleanup)

Remove from the new repo's `.repo-template.yml`:
- The `spec_test_map` entry referencing the playground test.
- The `extra_top_level_dirs: [mergepath, packaging]` entry.

Implement as a yq edit on the destination file (or a sed if yq isn't a hard dep — check what mergepath already requires).

### 4. Cross-repo loop update (step 5 of #156's flow — Mergepath-side write)

This step writes to MERGEPATH itself (not the new repo), so the wizard must:
- Open a NEW BRANCH on mergepath (`bootstrap/add-<new-repo>-to-loops`).
- Append `<new-repo>` to the documented loops in `DEPLOYMENT.md` (bootstrap loop + return-to-main loop) and `REVIEW_POLICY.md` (SSH-remote-switch loop).
- Commit, push, open a PR. Print the PR URL in the wizard summary.

The user's ASK in #156 says "the only mergepath-side write." Treating the cross-repo loop addition as a separate PR (not mixed with the new repo's bootstrap) keeps each PR's intent clean. The wizard prompts before opening the PR ("This will open a PR on mergepath itself to add `<new-repo>` to the cross-repo loops. Proceed? [y/N]").

### 5. Test coverage

`scripts/ci/check_bootstrap_template_mirror`:
- Stub a destination directory; run the mirror step against a fixture mergepath snapshot.
- Verify all 12 exclude paths are honored (assert each excluded path is absent from the destination).
- Verify name-bearing files are substituted correctly (no remaining `{{REPO_NAME}}` markers).
- Verify intentionally-unchanged files (e.g., `ai_agent_tooling_standard.md`, the `mergepath#NN` historical refs in scripts) are NOT substituted.
- Verify `.repo-template.yml` has the playground test + extra_top_level_dirs entries dropped.
- Verify the cross-repo loop update PR-creation step is dry-run-able (it doesn't actually open a PR in the test).

### 6. Bootstrap-mode markers

Adding `{{REPO_NAME}}` markers to the 6 mergepath files is itself a behavior change for direct mergepath operators (the markers shouldn't render unsubstituted in mergepath's own README). Two approaches:

a. **Sentinel pair**: use `<!-- BOOTSTRAP-MARKER -->{{REPO_NAME}}<!-- END -->` so mergepath's own rendering tools can hide the markers in mergepath's own context; the wizard strips the sentinels during substitution.

b. **Mergepath-as-itself substitution**: at mergepath bootstrap time (a one-shot migration), substitute `{{REPO_NAME}}` → `mergepath` in mergepath's own copies of these files. The markers then ONLY exist in the wizard's substitution table; mergepath's running files are clean.

Going with (b). One-shot migration in this sub-issue's PR.

## Acceptance

- [ ] `scripts/bootstrap/template-mirror.sh` ships, callable from sub-A's stage dispatch.
- [ ] All 12 documented exclude paths are honored.
- [ ] All 6 name-bearing files have `{{REPO_NAME}}`-style markers introduced + a one-shot migration applied to mergepath's own copies.
- [ ] `.repo-template.yml` has playground-specific entries dropped on substitution.
- [ ] Cross-repo loop update opens a separate Mergepath-side PR (not mixed with the new repo's bootstrap).
- [ ] `scripts/ci/check_bootstrap_template_mirror` test suite passes.
- [ ] On a real bootstrap (e.g., `./scripts/bootstrap-new-repo.sh test-repo --dry-run`), the mirror stage's plan output matches the matchline bootstrap walkthrough's actual file set.

## Out of scope

- GitHub repo creation, collaborators, labels, secrets — see #156 sub-C.
- Firebase/CodeRabbit/Codex App — see #156 sub-D.
- Project board + summary + runbook — see #156 sub-E.
- Wizard scaffold itself — see #156 sub-A.

Authoring-Agent: claude

### PR DESCRIPTION
**Pull request by nathanjohnpayne**
Closes #204.

## Summary

#156 sub-B implementation. The wizard's stage B (`template-mirror`) now ships the real implementation in place of sub-A's stub:

1. **rsync mergepath → target** with a curated exclude list (`dist/`, `mergepath/`, `packaging/`, `.claude/{worktrees,settings.local,launch}`, playground spec+plan+test, `scripts/policy-sim.sh`, screenshots dirs, prior `.bootstrap-{log,state}`). Single source of truth (`BOOTSTRAP_MIRROR_EXCLUDES`).
2. **Post-rsync orphan cleanup** (drop `tests/test_mergepath_playground.sh`, tombstone empty `bugs/`).
3. **Name substitution** (`scripts/bootstrap/substitute.sh`) across the 6 documented name-bearing files. Three case forms (`mergepath` / `Mergepath` / `MERGEPATH`) + URL. Per #204 acceptance § "Bootstrap-mode markers" option (b), no `{{REPO_NAME}}` markers introduced — keeps mergepath's own README readable without dual-rendering.
4. **`.repo-template.yml` cleanup** — yq drops `spec_test_map.mergepath_playground` + entire `extra_top_level_dirs` key.
5. **`git init` + initial commit** in target. `commit.gpgsign=false` scoped to the initial commit so machines without configured signing can still run the wizard.
6. **Cross-repo loop update** (anchor-driven, defensive). Probes for `<!-- bootstrap-loop-list-end -->` in DEPLOYMENT.md / REVIEW_POLICY.md. When absent (current state — separate doc-refactor PR introduces the anchors), logs a "manual action needed" warning and skips cleanly. When present, opens a Mergepath-side PR with the new repo appended to the loop list.

## Test plan

- [x] `bash tests/test_bootstrap_template_mirror.sh` → 29/29 PASS (excludes honored, orphan cleanup, substitution across all 3 case forms + URL, `.repo-template.yml` cleanup, git init, cross-repo loop step skips cleanly without anchors, --dry-run produces no side effects, substitute is idempotent)
- [x] `bash tests/test_bootstrap_wizard.sh` → 41/41 PASS (sub-A scaffold regression: only test 10's banner string needed updating)
- [x] All `scripts/ci/check_*` green (16 checks ran)
- [x] `bash -n` on all touched bash files

## Files

- New: `scripts/bootstrap/substitute.sh` (name-substitution lib)
- New: `scripts/ci/check_bootstrap_template_mirror` (CI wrapper)
- New: `tests/test_bootstrap_template_mirror.sh` (29-test fixture)
- Updated: `scripts/bootstrap/template-mirror.sh` (stub → full implementation, 6 steps)
- Updated: `scripts/bootstrap-new-repo.sh` (export `TARGET_DIR`, `BOOTSTRAP_MERGEPATH_ROOT`, `BOOTSTRAP_LIB_DIR`, `BOOTSTRAP_REPO_OWNER` so stage modules have a defined contract)
- Updated: `tests/test_bootstrap_wizard.sh` (banner expectation: "template-mirror (sub-B stub)" → "Starting stage: template-mirror")

## Out of scope (followups)

- **Anchor introduction in DEPLOYMENT.md + REVIEW_POLICY.md.** The cross-repo loop step is anchor-driven; the current docs use bash-embedded `for repo in ...` lists rather than structured lists. A separate doc-refactor PR converts those to anchored bullet lists so the wizard's loop step can find the insertion point. This is intentionally deferred — refactoring the loop docs is its own concern with its own review surface.
- #156 sub-C/D/E (GitHub infra, Firebase + CodeRabbit + Codex App, board + summary + runbook).

Authoring-Agent: claude

## Self-Review

- **Correctness:** 29-test fixture exercises every documented behavior. Substitution lib is tested both via the stage and directly via `bootstrap::_substitute_one_file`. Edge cases covered: no-match content preservation, idempotency.
- **Regression risk:** sub-A scaffold tests still pass; only the stub-banner string needed updating in `test_bootstrap_wizard.sh` test 10.
- **Style:** matches sub-A's helper conventions; every side-effect goes through `bootstrap::run` so `--dry-run` is correct.
- **Test coverage:** dry-run path verified to produce no on-disk side effects. Cross-repo loop step verified to leave the source worktree untouched when anchors absent.
- **Security:** rsync runs over local paths only. Substitution sed-escapes replacement-side input (the new repo name passes through `${var//&/\\&}` etc.). `commit.gpgsign=false` is scoped to the initial commit only — operator's regular signing applies to subsequent work.

### CODE REVIEW

File: scripts/bootstrap/template-mirror.sh [hunk 1]
```diff
@@ -1,27 +1,436 @@
...
+
+  # Step 5: initialize git history.
+  bootstrap::_init_target_git "$target"
+
+  # Step 6: cross-repo loop update. Writes to mergepath itself, so
+  # gated on a confirmation prompt.
+  bootstrap::_cross_repo_loop_update "$source_root"
+
   bootstrap::record_stage "template-mirror"
   return 0
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 9dbef2e. Each step's rc is now captured into a local `step_rc` and the stage early-returns on the first non-zero with a diagnostic identifying which step failed:
```bash
local step_rc=0
bootstrap::_rsync_template "$source_root" "$target" || step_rc=$?
if [ "$step_rc" -ne 0 ]; then
  bootstrap::err "template-mirror: rsync step failed (rc=$step_rc); aborting stage"
  return "$step_rc"
fi
# ... same shape for each subsequent step ...
bootstrap::record_stage "template-mirror"
return 0
```
`record_stage` is now reachable only on the all-steps-clean path. New test 32 simulates a failure via a yq shim that exits 1: stage returns non-zero AND the state file carries no `template-mirror` entry, so `--resume` can retry. Also extended `BOOTSTRAP_SKIP_MERGEPATH_GUARD` to gate the stage's own defense-in-depth re-check of mergepath state, mirroring the preflight skip.

File: scripts/bootstrap/template-mirror.sh [hunk 2]
```diff
@@ -1,27 +1,436 @@
...
+}
+
+bootstrap::_yq_clean_repo_template() {
+  local f=$1
+  # Drop the playground spec_test_map entry (whose key is
+  # "mergepath_playground" pre-substitution; substitution would
+  # have renamed it to e.g. "newrepo_playground" — drop either form
+  # by removing any entry whose value list contains the playground
+  # test path).
+  yq -i 'del(.spec_test_map.mergepath_playground)' "$f"
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/bootstrap/template-mirror.sh:273]: Addressed in 9dbef2e. Same bug CodeRabbit flagged on line 116 — swapped the call order so `_clean_repo_template_yml` runs BEFORE `apply_name_substitutions`. The yq `del(.spec_test_map.mergepath_playground)` now finds the deterministic pre-substitution key. New test 31 source-grep-asserts the call ordering so a future refactor that swaps them back fails loudly.

File: scripts/bootstrap/template-mirror.sh [hunk 3]
```diff
@@ -1,27 +1,436 @@
...
+  bootstrap::_rsync_template "$source_root" "$target"
+
+  # Step 2: post-mirror orphan cleanup.
+  bootstrap::_remove_orphans "$target"
+
+  # Step 3: apply name substitutions across the 6 name-bearing files.
+  bootstrap::apply_name_substitutions "$target"
+
+  # Step 4: drop mergepath-specific .repo-template.yml entries.
+  bootstrap::_clean_repo_template_yml "$target"
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 9dbef2e. Took your suggested fix verbatim — swapped the call order so `_clean_repo_template_yml` runs BEFORE `apply_name_substitutions`. The cleanup now operates on the deterministic original key names. Test 31 in `tests/test_bootstrap_template_mirror.sh` pins the ordering via source-grep so a future swap-back regression fails loudly.

File: scripts/bootstrap/template-mirror.sh [hunk 4]
```diff
@@ -1,27 +1,485 @@
...
+  local doc=$1 anchor=$2 line=$3
+  local tmp
+  tmp=$(mktemp "${TMPDIR:-/tmp}/bootstrap-loop.XXXXXX")
+  # awk: when we hit the anchor line, emit $line first, then the anchor.
+  awk -v anchor="$anchor" -v line="$line" '
+    $0 ~ anchor && !inserted { print line; inserted = 1 }
+    { print }
+  ' "$doc" > "$tmp"
+  mv "$tmp" "$doc"
+}
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 5fbcd27. Added a RETURN trap to clean up the mktemp'd tempfile and split the pipeline so awk/mv failures surface as the function's exit code:
```bash
local tmp
tmp=$(mktemp ...)
trap 'rm -f "$tmp"' RETURN
if ! awk ... > "$tmp"; then return 1; fi
if ! mv "$tmp" "$doc"; then return 1; fi
```

File: scripts/bootstrap/template-mirror.sh [hunk 5]
```diff
@@ -1,27 +1,485 @@
...
+
+bootstrap::_appendln() {
+  local doc=$1 line=$2
+  printf '%s\n' "$line" >> "$doc"
+}
```
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 5fbcd27. `bootstrap::_appendln` was a leftover from an earlier draft (round-0) where the cross-repo loop step would append-at-end-with-warning when anchors were absent. The current strategy is skip-with-warning (defer doc edits to a separate refactor PR), so the helper had no callers. Removed.

**Comment by nathanpayne-claude** [comment]
@codex review

**Comment by nathanpayne-claude** [comment]
## External Review Required

**PR:** #233 — feat(bootstrap): template mirror + name substitution + git init (#204)
**Branch:** feat/bootstrap-template-mirror-and-substitutions
**Author Agent:** claude

### Summary
#156 sub-B implementation. Stage B (`template-mirror`) ships the real implementation in place of sub-A's stub: rsync-with-excludes mirror of mergepath into the new repo, post-rsync orphan cleanup, name substitution across 6 documented files, `.repo-template.yml` cleanup, `git init` + initial commit, and an anchor-driven cross-repo loop update (defers cleanly when anchors absent). 1058 lines added across 6 files. Round 1 (Codex P1 + CodeRabbit Major + Codex P2) + round 2 (2 CodeRabbit trivial nits) addressed. 32/32 new tests + 41/41 sub-A regression tests pass.

### Focus Areas
- **Step-ordering invariant** (`scripts/bootstrap/template-mirror.sh:106-185`, `stage_template_mirror`): `_clean_repo_template_yml` MUST run before `apply_name_substitutions`. Round 1 caught this — the yq delete on `mergepath_playground` would have missed had substitution renamed the key first. Test 31 pins the ordering via source-grep. Worth scrutinizing whether other downstream consumers of this seam (sub-C/D/E) might re-introduce a similar ordering issue.
- **Failure propagation in the stage** (`stage_template_mirror`, the `step_rc` pattern): every step's rc is captured into a local; first non-zero short-circuits with a diagnostic. `record_stage` only runs on the all-clean path. Codex caught this on round 1 (the wizard's dispatch uses `"$fn" || stage_rc=$?` which disables `set -e` inside the called function). Verify the pattern is consistent — every step uses `|| step_rc=$?` and a follow-up `if [ "$step_rc" -ne 0 ]` block.
- **Substitution lib** (`scripts/bootstrap/substitute.sh`): three case forms (lowercase / Titlecased / UPPERCASE-with-hyphen-to-underscore) plus URL rewrite. Sed-escapes replacement-side input (`${var//&/\\&}` etc.). Worth checking for case-conflict pathologies — e.g., a filename containing both `Mergepath` and `mergepath` in the same line gets each form replaced correctly because sed applies all `-e` expressions to each line in order.
- **Cross-repo loop update** (`_cross_repo_loop_update`): defensively prompts, re-verifies mergepath state (now honoring `BOOTSTRAP_SKIP_MERGEPATH_GUARD=1`), and detects the `<!-- bootstrap-loop-list-end -->` anchor before opening a branch. When anchors are absent (current state — separate doc-refactor PR introduces them), it warns and skips without creating a stray branch or empty commit. The tempfile in `_anchor_insert` now has a `trap ... RETURN` cleanup (CodeRabbit round 2 nit).
- **Test 32** (yq-shim simulating sub-step failure): verifies the stage returns non-zero AND state file carries no completion entry — the resume mechanism's correctness depends on this.

### Observations from Internal Review
- The cross-repo loop step intentionally short-circuits when anchors are absent. The DEPLOYMENT.md + REVIEW_POLICY.md anchor introduction is deferred to a follow-up PR — that's a doc refactor with its own review surface, not a code change.
- `commit.gpgsign=false` is scoped to the target's initial commit only. The wizard's own commits and operator's day-to-day work keep regular signing.
- Codex App was unresponsive across multiple `@codex review` triggers over a 16+ minute window — that's why this is routing through 4b instead of 4a.
- Round 1 + 2 review history is in-thread on the PR; addressed findings have a code anchor (commit SHA) per reply.

### Suggested External Reviewer
nathanpayne-codex (CLI) or nathanpayne-cursor

### Rationale for External Review
1058 lines added (well above `external_review_threshold: 300`). Doesn't touch `external_review_paths` (no `src/auth/**`, `src/payments/**`, `**/*secret*`, etc.), but the line count alone gates it.

**Review by nathanpayne-codex — CHANGES_REQUESTED** [review]
Reviewed current head `5fbcd27` for PR #233.

Requesting changes for one runtime contract issue.

[P1] `scripts/bootstrap/template-mirror.sh` silently succeeds when `yq` is unavailable, leaving stale generated `.repo-template.yml` metadata. `_clean_repo_template_yml` currently warns and returns 0 when `command -v yq` fails, but the wizard preflight does not require `yq`. On a machine without `yq`, stage B exits 0 and records `template-mirror` as complete, then substitution rewrites the stale template-only keys instead of removing them. My no-`yq` fixture produced:

```yaml
spec_test_map:
  my-new-repo_playground:
    - tests/test_my-new-repo_playground.sh
  keep:
    - tests/test_keep.sh
extra_top_level_dirs: [my-new-repo, packaging]
```

That violates this stage's core cleanup contract and can generate a repo with a nonexistent playground test mapping plus incorrect top-level-dir allowances while reporting success. Please either make `yq` a hard dependency before the stage can run, or add a fallback cleanup path that removes these entries without `yq`; add a regression that hides `yq` from PATH and asserts the wizard does not silently complete with stale `.repo-template.yml` content.

Validation run on `5fbcd27`:

- `bash tests/test_bootstrap_template_mirror.sh` passed 32/32.
- `bash tests/test_bootstrap_wizard.sh` passed 41/41.
- all `scripts/ci/check_*` passed, with only existing repo-lint advisory warnings.
- `git diff --check origin/main...HEAD` passed.
- `bash -n scripts/bootstrap-new-repo.sh scripts/bootstrap/_lib.sh scripts/bootstrap/template-mirror.sh scripts/bootstrap/substitute.sh scripts/bootstrap/github-infra.sh scripts/bootstrap/firebase-and-codereview.sh scripts/bootstrap/board-and-summary.sh scripts/ci/check_bootstrap_template_mirror tests/test_bootstrap_template_mirror.sh tests/test_bootstrap_wizard.sh` passed.
- `scripts/coderabbit-wait.sh 233 nathanjohnpayne/mergepath` cleared current HEAD with 0 Potential issue markers.

I also verified the older Codex P1/P2 thread fixes in current code and resolved those two addressed bot threads. This review is blocked only on the missing-`yq` success path above.

**Comment by nathanpayne-claude** [comment]
@nathanpayne-codex — addressed in `ba2a9a1`. Two-layer fix per your finding:

**1. Wizard preflight gate (`scripts/bootstrap-new-repo.sh`, step 1)** — `yq` and `rsync` are now in the required-tools loop:
```bash
for tool in gh op git yq rsync; do ...
```
Plus a mikefarah/yq vs. kislyuk/yq detection (the Python wrapper has different syntax and would parse-fail mid-stage). Mirrors the check `scripts/sync-to-downstream.sh` already runs.

**2. Stage defense-in-depth (`bootstrap::_clean_repo_template_yml`)** — replaced the soft `warn` + `return 0` with `bootstrap::err` + `return 2`. The preflight gate is the primary defense; this keeps the stage fail-closed if invoked outside the wizard (future library re-use).

Tests (37/37 sub-B + 41/41 sub-A passing):
- **Assertion 13:** source the helper directly with yq absent from PATH. Verifies the diagnostic mentions "yq is required", rc=2, AND the .repo-template.yml is unmodified (no half-write).
- **Assertion 14:** invoke the whole wizard with a manufactured PATH containing every required tool EXCEPT yq. Verifies rc != 0 + the preflight diagnostic "missing required dependency: yq".

Ready for re-review on `ba2a9a1`.

**Review by nathanpayne-codex — CHANGES_REQUESTED** [review]
Reviewed current head `ba2a9a1` for PR #233.

The round-3 yq fix addresses my previous P1: `yq`/`rsync` are now preflight requirements, non-mikefarah yq is rejected, `_clean_repo_template_yml` fails closed with rc=2 when yq is unavailable, and the new missing-yq regressions cover both the helper and wizard preflight paths.

Requesting changes for two current-head blockers in the cross-repo loop step:

[P1] `_cross_repo_loop_update` still swallows failures from its internal side-effect steps. `stage_template_mirror` invokes it as `bootstrap::_cross_repo_loop_update "$source_root" || step_rc=$?`; under bash, that disables `errexit` inside the called function. The helper then runs `bootstrap::run` calls and direct `_append_repo_to_loop_doc` calls without capturing their return codes, so a failed checkout, commit, push, PR create, or doc insert can be masked by a later successful command such as returning to `main`. That can record `template-mirror` as complete even though the Mergepath loop PR was not actually opened. Please apply the same explicit rc-capture discipline inside `_cross_repo_loop_update` that the outer stage now uses, including preserving a cleanup/return-to-main path without masking the original failure.

[P1] The `gh pr create` write path still runs under whatever gh account is active. Per `REVIEW_POLICY.md`, author-identity writes (`gh pr create`, merge, label edit) must switch to `nathanjohnpayne` and then switch back. The wizard preflight says stage modules will switch as needed, but this stage does not. If an agent identity is active, the generated Mergepath loop PR will be authored by the reviewer account instead of `nathanjohnpayne`, which breaks the repo identity model and audit trail. Please wrap the PR-create step with a temporary switch to `nathanjohnpayne`, restore the prior active account even on failure, and propagate the original `gh pr create` rc.

Validation run on `ba2a9a1`:

- `bash tests/test_bootstrap_template_mirror.sh` passed 37/37.
- `bash tests/test_bootstrap_wizard.sh` passed 41/41.
- all `scripts/ci/check_*` passed, with only existing repo-lint advisory warnings.
- `git diff --check origin/main...HEAD` passed.
- `bash -n scripts/bootstrap-new-repo.sh scripts/bootstrap/_lib.sh scripts/bootstrap/template-mirror.sh scripts/bootstrap/substitute.sh scripts/bootstrap/github-infra.sh scripts/bootstrap/firebase-and-codereview.sh scripts/bootstrap/board-and-summary.sh scripts/ci/check_bootstrap_template_mirror tests/test_bootstrap_template_mirror.sh tests/test_bootstrap_wizard.sh` passed.
- `scripts/coderabbit-wait.sh 233 nathanjohnpayne/mergepath` exits 2 on this HEAD because of the two current CodeRabbit Potential issue threads above.

File: scripts/bootstrap/template-mirror.sh [hunk 6]
```diff
@@ -1,27 +1,504 @@
...
+- Correctness: anchor-driven insertion; falls back to append-at-end with warning if anchors missing.
+- Regression risk: low; pure doc append.
+- Style: matches existing entries.
+- Test coverage: scripts/ci/check_bootstrap_template_mirror covers the dry-run path.
+- Security: no new attack surface.
+"
+
+  # Switch back to main so the operator's worktree is left tidy.
+  bootstrap::run "return mergepath to main" \
+    git -C "$source_root" checkout -q main
```
[the remark this answers is not in the document]
Comment (nathanpayne-codex) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 008c343. Wrapped `gh pr create` with the author-identity switch-around per CLAUDE.md:
1. Capture prior active via `gh config get -h github.com user`.
2. `gh auth switch -u "$author_identity"` (default nathanjohnpayne; tests override via `BOOTSTRAP_AUTHOR_IDENTITY`).
3. Run `gh pr create` and capture rc into `$pr_create_rc`.
4. ALWAYS `gh auth switch -u "$prior_active"` (the switch-back runs regardless of pr-create success — slightly-off keyring is recoverable; obscured pr-create failure is not).
5. Surface `$pr_create_rc` as the function's failure code.

`BOOTSTRAP_SKIP_AUTHOR_SWITCH=1` is the test-only escape; real runs always switch. Assertion 16 in `tests/test_bootstrap_template_mirror.sh` source-grep-pins the switch → create → restore ordering.

File: scripts/bootstrap/template-mirror.sh [hunk 7]
```diff
@@ -1,27 +1,504 @@
...
+
+Authoring-Agent: claude
+
+## Self-Review
+- Correctness: anchor-driven insertion; falls back to append-at-end with warning if anchors missing.
+- Regression risk: low; pure doc append.
+- Style: matches existing entries.
+- Test coverage: scripts/ci/check_bootstrap_template_mirror covers the dry-run path.
+- Security: no new attack surface.
+"
```
[the remark this answers is not in the document]
Comment (nathanpayne-codex) [reply scripts/bootstrap/template-mirror.sh]: Addressed in 008c343 — same fix as the line 447 thread. The gh pr create call now runs inside a captured-prior-active / switch-to-author-identity / capture-rc / always-restore wrap per CLAUDE.md § Active-account convention. Restore runs even on pr-create failure so the operator's keyring is never left on the author identity.

**Comment by nathanpayne-codex** [comment]
@nathanpayne-codex — both blockers addressed in `008c343`.

**[P1] `_cross_repo_loop_update` failure swallowing.** Applied explicit rc-capture discipline inside the helper:
- Every `bootstrap::run` is now `|| step_rc=$?` with a conditional return-on-failure.
- Anchor-insert helpers (`bootstrap::_append_repo_to_loop_doc`) also captured + checked.
- Recovery paths (return-to-main, branch -D) on failure are best-effort and do NOT override the original failure rc.

**[P1] `gh pr create` author-identity switch-around.** Wrapped the PR create with `gh config get` → `gh auth switch -u $author_identity` → `gh pr create` (rc into `$pr_create_rc`) → ALWAYS `gh auth switch -u $prior_active`. The switch-back runs on both success and failure paths; obscured pr-create failure rc is surfaced after the restore. `$BOOTSTRAP_AUTHOR_IDENTITY` env var lets tests override the identity; `BOOTSTRAP_SKIP_AUTHOR_SWITCH=1` is the test-only bypass.

New tests (39/39 sub-B + 41/41 sub-A):
- **Assertion 15:** source-grep counts `bootstrap::run` calls inside `_cross_repo_loop_update` and ensures each has a paired `|| step_rc=$?` or `|| pr_create_rc=$?`.
- **Assertion 16:** source-grep verifies the switch → create → restore ordering, with a `^[[:space:]]*gh pr create ` anchor that distinguishes the actual command line from error messages mentioning "gh pr create".

Ready for re-review on `008c343`.

**Review by nathanpayne-codex — APPROVED** [review]
Reviewed current HEAD `008c343`.

The round-4 fixes address my blocking findings:

- `_cross_repo_loop_update` no longer relies on `set -e` inside the caller `|| step_rc=$?` context. Each side-effecting `bootstrap::run`/insert step now captures rc and returns non-zero on failure; recovery cleanup remains best-effort and does not mask the original rc.
- `gh pr create` now runs through the expected author-identity switch-around: capture prior active account, switch to `nathanjohnpayne` by default, capture `pr_create_rc`, always attempt restore, then return the PR-create rc.

Validation run locally on `008c343`:

- `bash tests/test_bootstrap_template_mirror.sh` — 39/39 passing
- `bash tests/test_bootstrap_wizard.sh` — 41/41 passing
- `scripts/ci/check_*` — passing, with only existing advisory warnings
- `git diff --check origin/main...HEAD`
- `bash -n` on the touched bootstrap/test scripts

CodeRabbit wait cleared on `008c343` with status success and 0 current-head potential issues. Live review threads are resolved/no unresolved blockers.

File: tests/test_bootstrap_template_mirror.sh [hunk 8]
```diff
@@ -0,0 +1,636 @@
...
+  # capture the helper rc into a variable and echo it. -u + pipefail
+  # are still desirable safety nets but -e would exit the subshell
+  # before reaching the echo on a non-zero rc.
+  set +e
+  BOOTSTRAP_DRY_RUN=0
+  BOOTSTRAP_LOG_FILE=""
+  bootstrap::_clean_repo_template_yml "$TARGET"
+  echo "RC=$?"
+' 2>&1)
+no_yq_ec=$?
```
[the remark this answers is not in the document]
Comment (nathanpayne-codex) [reply tests/test_bootstrap_template_mirror.sh:527]: Acknowledged. This is a valid low-value cleanup nit, but it is not blocking this external-review pass: the test already asserts the helper rc via the captured output, and pushing a one-line cleanup would churn the approved head without changing behavior. Leaving the approved HEAD unchanged.
